controllers/akodeploymentconfig/user/user_controller.go

```diff
@@ -327,18 +331,55 @@ func (r *AkoUserReconciler) ensureAkoUserRole() (*models.Role, error) {
 	if err != nil {
 		return role, err
 	}
-	// check if role needs to be sync
-	needSync := false
+
+	// check if role needs to be synced
+	if syncAkoUserRole(role) {
+		return r.aviClient.RoleUpdate(role)
+	}
+
+	return role, nil
+}
+
+// syncAkoUserRole checks the Role for the complete/correct
+// set of permissions and makes any additions/updates necessary.
+// Any additional permissions on the Role that are not part of
+// the desired AKO role are left as-is. It returns a bool
+// indicating whether the Role was changed.
+func syncAkoUserRole(role *models.Role) bool {
+	existingResources := sets.NewString()
```

Comment:
`sets.NewString()` is deprecated
```suggestion
	existingResources := sets.New[string]()
```